crates/provider/src/shared/types.rs

```diff
@@ -231,6 +267,8 @@ impl RunScriptOptions {
 pub struct TransferedFile {
     pub local_path: PathBuf,
     pub remote_path: PathBuf,
+    // TODO: Can be narrowed to have strict typing on this?
```

Comment:
Maybe using https://github.com/sourcefrog/unix_mode ?

Reply:
The issue is that it only allow to converts a mode as string, no the other way and I think that's what we would expect, user would pass string input and we convert it into a mode ? Maybe a custom type ?